### PR TITLE
Update Infra substring usage

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/infra/pull/404.

Please double-check this thoroughly as I have proven that this is hard for me :).

Will not build (so do not merge) until https://github.com/whatwg/infra/pull/404 is merged and given a bit of time to propagate into the linking databases.

***
<a href="https://pr-preview.s3.amazonaws.com/domenic/service-worker-scope-pattern-matching/pull/137.html" title="Last updated on Sep 21, 2021, 8:50 PM UTC (e971524)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/urlpattern/137/cdd4513...domenic:e971524.html" title="Last updated on Sep 21, 2021, 8:50 PM UTC (e971524)">Diff</a>

### CODE REVIEW

**Comment by wanderview** [comment]
Can you summarize what changed in infra?  Looks like its basically exclusive instead of inclusive now?

**Comment by domenic** [comment]
Yeah basically. It's more like JS `.substring()` now.

**Review by wanderview — APPROVED** [review]
LGTM

**Comment by wanderview** [comment]
@domenic, the upstream PR merged, but this still fails to build.

**Comment by domenic** [comment]
Yeah, it usually takes a day or two for the Bikeshed linking database to update.
